compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -307,6 +307,7 @@ fn mir_const(tcx: TyCtxt<'_>, def: LocalDefId) -> &Steal<Body<'_>> {
             &Lint(check_packed_ref::CheckPackedRef),
             &Lint(check_const_item_mutation::CheckConstItemMutation),
             &Lint(function_item_references::FunctionItemReferences),
+            &coroutine::ByMoveBody,
```

Comment:
why can we only do this here? we lack the information at creation time? (needs borrowck info)

is there a way to generate the body here instead of as a shim that gets then changed here?

Reply:
No, we can do it during build.

Reply:
Actually, it's easier to keep it here because we get automatic validation from the pass manager. I'll leave a comment explaining that it could be uplifted into mir_built or possibly generated on-demand rather than eagerly.